pyfar/classes/audio.py

```diff
@@ -691,7 +738,12 @@ def _freq(self, value, raw):
         self._domain = 'freq'
         if not raw:
             # remove normalization
-            data = fft.normalization(
+            if self.complex:
```

Comment:
self._n_samples is set a couple of lines above without checking self.complex. It should be done as in Signal.__init__. To reduce code redundancy we could add a private Class method `self._get_n_samples_from_frequency_data(self, data.shape[-1])`

Reply:
Good point. I check n_samples now and put everything in a private method.